src/controller.ts

```
@@ -0,0 +1,147 @@
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore - Binds is missing from ContainerCreateOptions type
+  // the as unknown as Docker.Container is also related to this mess
+  // TODO make a pull request there to add the Binds type
+  // https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/dockerode
+  Binds: [
+    '/var/lib/docker/volumes/volume_vm/_data/in:/app/in',
+    '/var/lib/docker/volumes/volume_vm/_data/out:/app/out',
+    '/var/lib/docker/volumes/volume_vm/_data/work:/app/work',
+  ],
+})) as unknown as Docker.Container;
+await createdContainer.start({});
+
+// eslint-disable-next-line no-console
+console.log('Container created with ID:', createdContainer.id);
+
+let timer: SetIntervalAsyncTimer;
+
+async function stopStats() : Promise<void> {
+  await clearIntervalAsync(timer);
+}
+
+export async function parseStats() : Promise<void> {
+  const directoryName = targetDirectory;
+  const fileList = await fsAsync.readdir(directoryName);
+
+  // Load all the files in parallel
+  const stats = await Promise.all(fileList
+    // Only load files following the stats.${counter}.json pattern
+    .filter((fileName) => /^stats\.\d+\.json$/.test(fileName))
+    .map(async (fileName) => {
+      const fullFilename = path.join(directoryName, fileName);
+      const data = await fsAsync.readFile(fullFilename, { encoding: 'utf-8' });
+
+      try {
+        const fileContent = JSON.parse(data) as {
+          read: string;
+          cpu_stats: {
+            cpu_usage: {
+              total_usage: number;
+            };
+          };
+          memory_stats: {
+            usage: number;
+            max_usage: number;
+          };
+          networks: {
+            eth0: {
+              rx_bytes: number;
+            };
+          };
+        };
+        if (fileContent.read.startsWith('0001-01-01T00:00:00Z')) {
+          throw new Error('Invalid date');
```

In general I recommend to throw errors instead of writing a message on the console and ignoring the problem. The error can be handled correctly, and if is not its message will be written on the console.

If you want to ignore errors, it's fine, but please write a comment about that or a try/catch block :)